lib/jar.ts

```diff
@@ -0,0 +1,26 @@
+import * as path from 'path';
+import * as crypto from 'crypto';
+import * as fs from 'fs';
+
+// Using the maven-central sha1 checksum API call (see: https://search.maven.org/classic/#api)
+const SHA1 = crypto.createHash('sha1');
```

Comment:
this needs to be scoped in getSha1ForJar function not in the module as we need to create a new Hash object on each call

add another call to getSha1ForJar in the test to see the effect, I think it will error.

Reply:
we can have

```
const HASH_ALGORITHM = 'sha1';
const DIGEST_ENCODING = 'hex';
```

at this level though